### ISSUE TITLE
"Waiting to start" for a long time, makeslice: cap out of range in the logs

### ISSUE THREAD

**Issue by reggie-k** [argoproj/argo-cd#5097]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [ ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* argocd version v1.8.1+c2547dc.

**Sometimes the sync takes long time to start, and in the logs of the controller I see:

time="2020-12-21T12:53:08Z" level=error msg="Recovered from panic: runtime error: makeslice: cap out of range\ngoroutine 188 [running]:\nruntime/debug.Stack(0xc020f78d40, 0x1e906a0, 0x24ad210)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x9d\ngithub.com/argoproj/argo-cd/controller.(*ApplicationController).processAppRefreshQueueItem.func1(0xc000384f00, 0x1de4bc0, 0xc0004f5b40)\n\t/go/src/github.com/argoproj/argo-cd/controller/appcontroller.go:1002 +0x88\npanic(0x1e906a0, 0x24ad210)\n\t/usr/local/go/src/runtime/panic.go:975 +0x3e3\ngithub.com/vmihailenco/bufpool.(*bufPool).Get(0x38a9e40, 0x200e429, 0xc020f78f80)\n\t/go/pkg/mod/github.com/vmihailenco/bufpool@v0.1.11/buf_pool.go:45 +0x273\ngithub.com/vmihailenco/bufpool.Get(...)\n\t/go/pkg/mod/github.com/vmihailenco/bufpool@v0.1.11/buf_pool.go:17\ngithub.com/vmihailenco/bufpool.(*Buffer).grow(0xc036658ab0, 0xe429, 0x1ba5813)\n\t/go/pkg/mod/github.com/vmihailenco/bufpool@v0.1.11/buffer_ext.go:57 +0x172\ngithub.com/vmihailenco/bufpool.(*Buffer).Write(0xc036658ab0, 0xc0345f4000, 0xe429, 0xe429, 0xc020f78fe8, 0x1ba6e9e, 0xc08b6bff20)\n\t/go/pkg/mod/github.com/vmihailenco/bufpool@v0.1.11/buffer.go:121 +0xdc\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).writeString(0xc08b6bff20, 0xc0345f4000, 0xe429, 0x0, 0x9)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode.go:256 +0x75\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).encodeNormalString(0xc08b6bff20, 0xc0345f4000, 0xe429, 0x20a3600, 0xc1009f46e0)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_slice.go:78 +0x7e\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).EncodeString(0xc08b6bff20, 0xc0345f4000, 0xe429, 0xc0345f4000, 0xe429)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_slice.go:71 +0xa5\ngithub.com/vmihailenco/msgpack/v5.encodeStringValue(0xc08b6bff20, 0x1de4bc0, 0xc1009f4730, 0x198, 0x1, 0x1)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_slice.go:13 +0x61\ngithub.com/vmihailenco/msgpack/v5.(*field).EncodeValue(0xc03a1fb140, 0xc08b6bff20, 0x20a3600, 0xc1009f46e0, 0x199, 0x0, 0x0)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/types.go:113 +0xa5\ngithub.com/vmihailenco/msgpack/v5.encodeStructValue(0xc08b6bff20, 0x20a3600, 0xc1009f46e0, 0x199, 0xc1009f46e0, 0x199)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_map.go:160 +0x214\ngithub.com/vmihailenco/msgpack/v5.ptrEncoderFunc.func1(0xc08b6bff20, 0x209d1a0, 0xc054fe81d0, 0x196, 0x209d1a0, 0x0)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_value.go:129 +0xa8\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).EncodeValue(0xc08b6bff20, 0x209d1a0, 0xc054fe81d0, 0x196, 0x209d1a0, 0xc054fe81d0)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode.go:228 +0x8c\ngithub.com/vmihailenco/msgpack/v5.encodeArrayValue(0xc08b6bff20, 0x1d80140, 0xc100fa3900, 0x97, 0x2285578, 0x1)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_slice.go:134 +0xe2\ngithub.com/vmihailenco/msgpack/v5.encodeSliceValue(0xc08b6bff20, 0x1d80140, 0xc100fa3900, 0x97, 0x1d80140, 0x7f1d305d3690)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_slice.go:125 +0x78\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).EncodeValue(0xc08b6bff20, 0x1d80140, 0xc100fa3900, 0x97, 0xc036658ab0, 0x2139301)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode.go:228 +0x8c\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).Encode(0xc08b6bff20, 0x1d80140, 0xc100fa3900, 0x0, 0x203018)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode.go:214 +0xe9\ngithub.com/go-redis/cache/v8.(*Cache).Marshal(0xc0001a2b40, 0x1d80140, 0xc100fa3900, 0x0, 0x0, 0x0, 0x0, 0x0)\n\t/go/pkg/mod/github.com/go-redis/cache/v8@v8.2.1/cache.go:308 +0x167\ngithub.com/go-redis/cache/v8.(*Cache).set(0xc0001a2b40, 0xc0625a1b30, 0xc0048f6000, 0x26, 0xc020f794c0, 0x40cd28, 0x30, 0x1f6d940)\n\t/go/pkg/mod/github.com/go-redis/cache/v8@v8.2.1/cache.go:129 +0x64\ngithub.com/go-redis/cache/v8.(*Cache).Set(...)\n\t/go/pkg/mod/github.com/go-redis/cache/v8@v8.2.1/cache.go:119\ngithub.com/argoproj/argo-cd/util/cache.(*redisCache).Set(0xc000717b60, 0xc063527470, 0xc020f79518, 0x2)\n\t/go/src/github.com/argoproj/argo-cd/util/cache/redis.go:33 +0xb0\ngithub.com/argoproj/argo-cd/util/cache.(*Cache).SetItem(0xc0004da150, 0xc099eaff50, 0x26, 0x1d80140, 0xc100fa3900, 0x34630b8a000, 0x0, 0xc100957500, 0x20)\n\t/go/src/github.com/argoproj/argo-cd/util/cache/cache.go:89 +0x164\ngithub.com/argoproj/argo-cd/util/cache/appstate.(*Cache).SetItem(...)\n\t/go/src/github.com/argoproj/argo-cd/util/cache/appstate/cache.go:51\ngithub.com/argoproj/argo-cd/util/cache/appstate.(*Cache).SetAppManagedResources(0xc0004da160, 0xc0009ef8c4, 0xa, 0xc054fe8000, 0x4a, 0x4a, 0x0, 0x0)\n\t/go/src/github.com/argoproj/argo-cd/util/cache/appstate/cache.go:64 +0x125\ngithub.com/argoproj/argo-cd/controller.(*ApplicationController).setAppManagedResources(0xc000384f00, 0xc0452cf000, 0xc006e462c0, 0x9, 0xc06b625c98, 0xc000a361e0)\n\t/go/src/github.com/argoproj/argo-cd/controller/appcontroller.go:279 +0x14a\ngithub.com/argoproj/argo-cd/controller.(*ApplicationController).processAppRefreshQueueItem(0xc000384f00, 0x203001)\n\t/go/src/github.com/argoproj/argo-cd/controller/appcontroller.go:1088 +0x8c3\ngithub.com/argoproj/argo-cd/controller.(*ApplicationController).Run.func3()\n\t/go/src/github.com/argoproj/argo-cd/controller/appcontroller.go:474 +0x36\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000360970)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:155 +0x5f\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000360970, 0x24d29e0, 0xc000476000, 0x1, 0xc0004f8540)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:156 +0xa3\nk8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000360970, 0x3b9aca00, 0x0, 0x1, 0xc0004f8540)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:133 +0x98\nk8s.io/apimachinery/pkg/util/wait.Until(0xc000360970, 0x3b9aca00, 0xc0004f8540)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:90 +0x4d\ncreated by github.com/argoproj/argo-cd/controller.(*ApplicationController).Run\n\t/go/src/github.com/argoproj/argo-cd/controller/appcontroller.go:473 +0x44d\n"
**

**To Reproduce**

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue.

**Immediate sync like in other cases **

**Screenshots**

**Version**

```shell
v1.8.1+c2547dc
```

**Logs**

```
time="2020-12-21T12:59:20Z" level=error msg="Recovered from panic: runtime error: makeslice: cap out of range\ngoroutine 144 [running]:\nruntime/debug.Stack(0xc0661fad40, 0x1e906a0, 0x24ad210)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x9d\ngithub.com/argoproj/argo-cd/controller.(*ApplicationController).processAppRefreshQueueItem.func1(0xc000348780, 0x1de4bc0, 0xc0f88b7fa0)\n\t/go/src/github.com/argoproj/argo-cd/controller/appcontroller.go:1002 +0x88\npanic(0x1e906a0, 0x24ad210)\n\t/usr/local/go/src/runtime/panic.go:975 +0x3e3\ngithub.com/vmihailenco/bufpool.(*bufPool).Get(0x38a9e40, 0x20120b3, 0xc0661faf80)\n\t/go/pkg/mod/github.com/vmihailenco/bufpool@v0.1.11/buf_pool.go:45 +0x273\ngithub.com/vmihailenco/bufpool.Get(...)\n\t/go/pkg/mod/github.com/vmihailenco/bufpool@v0.1.11/buf_pool.go:17\ngithub.com/vmihailenco/bufpool.(*Buffer).grow(0xc09972bfb0, 0x120b3, 0x1ba5937)\n\t/go/pkg/mod/github.com/vmihailenco/bufpool@v0.1.11/buffer_ext.go:57 +0x172\ngithub.com/vmihailenco/bufpool.(*Buffer).Write(0xc09972bfb0, 0xc06371a000, 0x120b3, 0x120b3, 0xc0661fafe8, 0x1ba6ed9, 0xc05fccca80)\n\t/go/pkg/mod/github.com/vmihailenco/bufpool@v0.1.11/buffer.go:121 +0xdc\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).writeString(0xc05fccca80, 0xc06371a000, 0x120b3, 0x0, 0x12)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode.go:256 +0x75\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).encodeNormalString(0xc05fccca80, 0xc06371a000, 0x120b3, 0x20a3600, 0xc0900503c0)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_slice.go:78 +0x7e\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).EncodeString(0xc05fccca80, 0xc06371a000, 0x120b3, 0xc06371a000, 0x120b3)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_slice.go:71 +0xa5\ngithub.com/vmihailenco/msgpack/v5.encodeStringValue(0xc05fccca80, 0x1de4bc0, 0xc090050448, 0x198, 0x1, 0x1)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_slice.go:13 +0x61\ngithub.com/vmihailenco/msgpack/v5.(*field).EncodeValue(0xc02d35cac0, 0xc05fccca80, 0x20a3600, 0xc0900503c0, 0x199, 0x0, 0x0)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/types.go:113 +0xa5\ngithub.com/vmihailenco/msgpack/v5.encodeStructValue(0xc05fccca80, 0x20a3600, 0xc0900503c0, 0x199, 0xc0900503c0, 0x199)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_map.go:160 +0x214\ngithub.com/vmihailenco/msgpack/v5.ptrEncoderFunc.func1(0xc05fccca80, 0x209d1a0, 0xc0322d4958, 0x196, 0x209d1a0, 0x0)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_value.go:129 +0xa8\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).EncodeValue(0xc05fccca80, 0x209d1a0, 0xc0322d4958, 0x196, 0x209d1a0, 0xc0322d4958)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode.go:228 +0x8c\ngithub.com/vmihailenco/msgpack/v5.encodeArrayValue(0xc05fccca80, 0x1d80140, 0xc0cfd757c0, 0x97, 0x2285578, 0xc0661fb201)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_slice.go:134 +0xe2\ngithub.com/vmihailenco/msgpack/v5.encodeSliceValue(0xc05fccca80, 0x1d80140, 0xc0cfd757c0, 0x97, 0x1d80140, 0x7f0cb492e5f0)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode_slice.go:125 +0x78\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).EncodeValue(0xc05fccca80, 0x1d80140, 0xc0cfd757c0, 0x97, 0xc09972bfb0, 0x2139301)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode.go:228 +0x8c\ngithub.com/vmihailenco/msgpack/v5.(*Encoder).Encode(0xc05fccca80, 0x1d80140, 0xc0cfd757c0, 0x0, 0x20300e)\n\t/go/pkg/mod/github.com/vmihailenco/msgpack/v5@v5.1.0/encode.go:214 +0xe9\ngithub.com/go-redis/cache/v8.(*Cache).Marshal(0xc000166900, 0x1d80140, 0xc0cfd757c0, 0x0, 0x0, 0x0, 0x0, 0x0)\n\t/go/pkg/mod/github.com/go-redis/cache/v8@v8.2.1/cache.go:308 +0x167\ngithub.com/go-redis/cache/v8.(*Cache).set(0xc000166900, 0xc039276d70, 0xc039286c00, 0x26, 0xc0661fb4c0, 0x40cd28, 0x30, 0x1f6d940)\n\t/go/pkg/mod/github.com/go-redis/cache/v8@v8.2.1/cache.go:129 +0x64\ngithub.com/go-redis/cache/v8.(*Cache).Set(...)\n\t/go/pkg/mod/github.com/go-redis/cache/v8@v8.2.1/cache.go:119\ngithub.com/argoproj/argo-cd/util/cache.(*redisCache).Set(0xc0001a6580, 0xc04506e4b0, 0xc0661fb518, 0x2)\n\t/go/src/github.com/argoproj/argo-cd/util/cache/redis.go:33 +0xb0\ngithub.com/argoproj/argo-cd/util/cache.(*Cache).SetItem(0xc0006b2b70, 0xc06c7d4c30, 0x26, 0x1d80140, 0xc0cfd757c0, 0x34630b8a000, 0x0, 0xc0b667b500, 0x20)\n\t/go/src/github.com/argoproj/argo-cd/util/cache/cache.go:89 +0x164\ngithub.com/argoproj/argo-cd/util/cache/appstate.(*Cache).SetItem(...)\n\t/go/src/github.com/argoproj/argo-cd/util/cache/appstate/cache.go:51\ngithub.com/argoproj/argo-cd/util/cache/appstate.(*Cache).SetAppManagedResources(0xc0006b2c00, 0xc000ca7984, 0xa, 0xc0322d4780, 0x4a, 0x4a, 0x0, 0x0)\n\t/go/src/github.com/argoproj/argo-cd/util/cache/appstate/cache.go:64 +0x125\ngithub.com/argoproj/argo-cd/controller.(*ApplicationController).setAppManagedResources(0xc000348780, 0xc07b2de400, 0xc02e394000, 0x7, 0xc036d43c98, 0xc000c29c70)\n\t/go/src/github.com/argoproj/argo-cd/controller/appcontroller.go:279 +0x14a\ngithub.com/argoproj/argo-cd/controller.(*ApplicationController).processAppRefreshQueueItem(0xc000348780, 0x203001)\n\t/go/src/github.com/argoproj/argo-cd/controller/appcontroller.go:1088 +0x8c3\ngithub.com/argoproj/argo-cd/controller.(*ApplicationController).Run.func3()\n\t/go/src/github.com/argoproj/argo-cd/controller/appcontroller.go:474 +0x36\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00049a0c0)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:155 +0x5f\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00049a0c0, 0x24d29e0, 0xc0000c6d80, 0x1, 0xc000109020)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:156 +0xa3\nk8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00049a0c0, 0x3b9aca00, 0x0, 0x1, 0xc000109020)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:133 +0x98\nk8s.io/apimachinery/pkg/util/wait.Until(0xc00049a0c0, 0x3b9aca00, 0xc000109020)\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:90 +0x4d\ncreated by github.com/argoproj/argo-cd/controller.(*ApplicationController).Run\n\t/go/src/github.com/argoproj/argo-cd/controller/appcontroller.go:473 +0x44d\n"

```

**Comment by alexmt** [comment]
This was fixed by https://github.com/argoproj/argo-cd/pull/4965 in master. Cherry-picking into 1.8

**Comment by alexmt** [comment]
Fix is available in v1.8.2: https://github.com/argoproj/argo-cd/releases/tag/v1.8.2
